packages/makeup-navigation-emitter/test/index.js

```diff
@@ -368,4 +368,270 @@ describe('given 3 items with default options', function() {
+
+        it('should set the index to 0', function() {
+            expect(testEmitter.model.index).toBe(0);
+        });
+    });
+
+    describe('when testEmitter is destroyed and recreated', function() {
+        beforeAll(function() {
+            // reset testEmitter to test onNavigationModelInit
+            testEmitter.destroy();
+
+            onNavigationModelInit = jasmine.createSpy('onNavigationModelInit');
+            testEl.addEventListener('navigationModelInit', onNavigationModelInit);
+
+            testEl = document.querySelector('.widget');
+            testEmitter = NavigationEmitter.createLinear(testEl, 'li', {autoInit: 1}); // eslint-disable-line
+        });
+
+        it('should trigger 1 navigationModelInit event', function() {
+            expect(onNavigationModelInit).toHaveBeenCalledTimes(1);
+        });
+    });
+});
+
+describe('given 3 items with autoReset set to 1', function() {
+    var onNavigationModelReset;
+    function setup() {
+        document.body.innerHTML = `
+            <ul class="widget">
+                <li>Item 1</li>
+                <li>Item 2</li>
+                <li>Item 3</li>
+            </ul>
+        `;
+
+        testEl = document.querySelector('.widget');
+        testEmitter = NavigationEmitter.createLinear(testEl, 'li', {autoReset: 1}); // eslint-disable-line
+
+        onNavigationModelChange = jasmine.createSpy('onNavigationModelChange');
+        testEl.addEventListener('navigationModelChange', onNavigationModelChange);
+    }
+
+    beforeAll(setup);
+    afterEach(setup);
+
+    describe('when arrow right is pressed once', function() {
+        beforeAll(function() {
+            triggerArrowKeyPress(testEl, 'Right', 1);
+        });
+
+        it('should set the index to 1', function() {
+            expect(testEmitter.model.index).toBe(1);
+        });
+    });
+
+    describe('when arrow up is pressed once', function() {
+        beforeAll(function() {
+            triggerArrowKeyPress(testEl, 'Up', 1);
+        });
+
+        it('should set the index to 0', function() {
+            expect(testEmitter.model.index).toBe(0);
+        });
+    });
+
+    describe('when testEmitter gets reset', function() {
+        beforeAll(function() {
+            onNavigationModelReset = jasmine.createSpy('onNavigationModelReset');
+            testEl.addEventListener('navigationModelReset', onNavigationModelReset);
+        });
+
+        it('should trigger 1 onNavigationModelReset event', function() {
+            testEmitter.model.reset();
+            expect(onNavigationModelReset).toHaveBeenCalledTimes(1);
+        });
+    });
+
+    // describe('when focus exits the widget', function() {
+    //     var eventHandlers = {
+    //         onFocusExit: function() {}
+    //     };
+
+    //     beforeAll(function() {
+    //         spyOn(eventHandlers, 'onFocusExit');
+    //         testEl.addEventListener('focusExit', eventHandlers.onFocusExit);
+
+    //         testEl.blur();
```

Comment:
@ianmcburnie, how can we test focus exit events?

Reply:
I did it here by moving focus onto a sibling element